clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -465,6 +465,188 @@ public void testSyncGroupRequestWithFencedInstanceIdException() {
         assertThrows(FencedInstanceIdException.class, () -> coordinator.ensureActiveGroup());
     }
 
+    @Test
+    public void testJoinGroupUnknownMemberResponseWithOldGeneration() {
```

Comment:
Could we extract the group setup logics until the `coordinator.ensureActiveGroup();`, they seem to be redundant to repeat a couple of times.

Reply:
Good point, will try it out.